pkg/reconciler/sinkbinding/sinkbinding.go

```diff
@@ -0,0 +1,196 @@
+	"knative.dev/pkg/webhook/psbinding"
+)
+
+type SinkBindingSubResourcesReconciler struct {
+	res                  *resolver.URIResolver
+	tracker              tracker.Interface
+	serviceAccountLister corev1listers.ServiceAccountLister
+	secretLister         corev1listers.SecretLister
+	kubeclient           kubernetes.Interface
+	featureStore         *feature.Store
+	tokenProvider        *auth.OIDCTokenProvider
+}
+
+func (s *SinkBindingSubResourcesReconciler) Reconcile(ctx context.Context, b psbinding.Bindable) error {
+	sb := b.(*v1.SinkBinding)
+	if s.res == nil {
+		err := errors.New("Resolver is nil")
+		logging.FromContext(ctx).Errorf("%w", err)
+		sb.Status.MarkBindingUnavailable("NoResolver", "No Resolver associated with context for sink")
+		return err
+	}
+	if sb.Spec.Sink.Ref != nil {
+		s.tracker.TrackReference(tracker.Reference{
+			APIVersion: sb.Spec.Sink.Ref.APIVersion,
+			Kind:       sb.Spec.Sink.Ref.Kind,
+			Namespace:  sb.Spec.Sink.Ref.Namespace,
+			Name:       sb.Spec.Sink.Ref.Name,
+		}, b)
+	}
+
+	addr, err := s.res.AddressableFromDestinationV1(ctx, sb.Spec.Sink, sb)
+	if err != nil {
+		logging.FromContext(ctx).Errorf("Failed to get Addressable from Destination: %w", err)
+		sb.Status.MarkBindingUnavailable("NoAddressable", "Addressable could not be extracted from destination")
+		return err
+	}
+	sb.Status.MarkSink(addr)
+
+	featureFlags := s.featureStore.Load()
+	if featureFlags.IsOIDCAuthentication() {
+		saName := auth.GetOIDCServiceAccountNameForResource(v1.SchemeGroupVersion.WithKind("SinkBinding"), sb.ObjectMeta)
+		sb.Status.Auth = &duckv1.AuthStatus{
+			ServiceAccountName: &saName,
+		}
+
+		if err := auth.EnsureOIDCServiceAccountExistsForResource(ctx, s.serviceAccountLister, s.kubeclient, v1.SchemeGroupVersion.WithKind("SinkBinding"), sb.ObjectMeta); err != nil {
+			sb.Status.MarkOIDCIdentityCreatedFailed("Unable to resolve service account for OIDC authentication", "%v", err)
+			return err
+		}
+		sb.Status.MarkOIDCIdentityCreatedSucceeded()
+
+		err := s.reconcileOIDCTokenSecret(ctx, sb)
+		if err != nil {
+			sb.Status.MarkOIDCTokenSecretCreatedFailed("Unable to reconcile OIDC token secret", "%v", err)
+			return err
+		}
+		sb.Status.MarkOIDCTokenSecretCreatedSuccceeded()
+
+	} else {
+		sb.Status.Auth = nil
+		sb.Status.MarkOIDCIdentityCreatedSucceededWithReason(fmt.Sprintf("%s feature disabled", feature.OIDCAuthentication), "")
+		sb.Status.MarkOIDCTokenSecretCreatedSuccceededWithReason(fmt.Sprintf("%s feature disabled", feature.OIDCAuthentication), "")
+		sb.Status.OIDCTokenSecretName = nil
+	}
+
+	return nil
+}
+
+// I'm just here so I won't get fined
+func (*SinkBindingSubResourcesReconciler) ReconcileDeletion(ctx context.Context, b psbinding.Bindable) error {
+	return nil
+}
+
+func (s *SinkBindingSubResourcesReconciler) reconcileOIDCTokenSecret(ctx context.Context, sb *v1.SinkBinding) error {
+	logger := logging.FromContext(ctx)
+	secretName := fmt.Sprintf("oidc-token-%s", sb.Name)
```

Comment:
This is potentially longer than the max characters count, can we just use the SB name?

Reply:
:+1: updated to use kmeta.ChildName()